Pokepedia.Api/Controllers/Pokemons/LocationGet/LocationByPokemonNameController.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Pokepedia.Api.Controllers.Pokemons.LocationGet.Requests;
+using Pokepedia.Api.Controllers.Pokemons.LocationGet.Responses;
+using Pokepedia.Domain.Entities.Pokemons;
+using Pokepedia.Domain.Services.Crud.Read;
+using Pokepedia.Domain.Validation;
+using Pokepedia.Domain.Validation.LocationValidation;
+
+namespace Pokepedia.Api.Controllers.Pokemons.LocationGet
+{
+    [ApiController]
+    [Route("v1")]
+    public class LocationByPokemonNameController : ControllerBase
+    {
+        private readonly ILogger<LocationByPokemonNameController> _logger;
+        private readonly IReadByNameServiceForLocation<PokemonName, Pokemon> _locationGetByPokemonNameService;
+    }
+
+    [HttpGet("location/getByName")]
+    public async Task<ActionResult<GetLocationByPokemonNameRequest>> HandleAsync([FromQuery] GetLocationByPokemonNameRequest getLocationByPokemonNameRequest, CancellationToken cancellation = default)
+    {
+        var locationToGet = new LocationByName(getLocationByPokemonNameRequest.Name);
+        var getLocation = await _locationGetByPokemonNameService.GetLocationByPokemonNameAsync(locationToGet);
+
+        var response = new GetLocationByPokemonNameResponse()
+        {
+            Name = getLocation.Name,
```

Comment:
This is a question not a command. But if we are getting the location for a pokemon right. By the name too. Would we expect the response to have the name of the pokemon we passed in? I could see the id but the name doesnt seem needed to me

Reply:
it just seemed helpful to have the name also displayed alongside the corresponding locations. No big deal to do away with it

Reply:
but now youre tacking on extra information the user didnt ask for. Its not that big of a deal. But these things snowball fast. So I tend to like being very strict

Reply:
true, i was just trying to account for the user wanting to see what pokemon they had entered in. But that doesnt matter with the current way the user types it in

Reply:
You seem to feel strongly about this so lets keep it in. We can remove it if we feel its not needed later.